packages/devextreme/js/__internal/scheduler/tooltip_strategies/m_tooltip_strategy_base.ts

```diff
@@ -14,6 +14,8 @@ const TOOLTIP_APPOINTMENT_ITEM_DELETE_BUTTON_CONTAINER = `${TOOLTIP_APPOINTMENT_
 const TOOLTIP_APPOINTMENT_ITEM_DELETE_BUTTON = `${TOOLTIP_APPOINTMENT_ITEM}-delete-button`;
 
 export class TooltipStrategyBase {
+  protected asyncTemplateDeferredList = new Set<Promise<void>>();
```

Comment:
Minor comment.
Why do we use ```Set``` here? :)
I didn't see any searching of the promises, can we use an array here instead?

I see the deletion of the promise from set by its ref, but we can do the following with the array easily:
```typescript
this.asyncTemplateRenderedArray = this.asyncTemplateRenderedArray.filter((item) => item !== promise);
```

Reply:
Because this filter operation in your example is the searching you're talking about
`set.remove(ref)` looks much better than `newArray = oldArray.filter(...)`
It's better practice to use sets if we don't depend on items order